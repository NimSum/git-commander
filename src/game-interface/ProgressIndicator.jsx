import React from 'react'

export function ProgressIndicator(props) {
  return (
    <div className="progress-indicator">
      <span
      className={props.currRound >= 6 && "current-round"}>Remote</span>
      <span
      className={props.currRound >= 5 && "current-round"}></span>
      <span
      className={props.currRound >= 4 && "current-round"}></span>
      <div className="vertical-line"></div>
      <span
      className={props.currRound >= 3 && "current-round"}></span>
      <span
      className={props.currRound >= 2 && "current-round"}></span>
      <span 
      className={props.currRound >= 1 && "current-round"}>Local</span>
    </div>
  )
}