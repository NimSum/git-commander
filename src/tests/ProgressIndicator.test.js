import React from 'react';
import ProgressIndicator from '../game-interface/ProgressIndicator';
import Adapter from 'enzyme-adapter-react-16';
import { shallow , configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Progress Indicator' , () => {

  let mockRound = 3;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      < ProgressIndicator 
        currRound={mockRound}
      />
    );
  });

  it('Should render card based on current round', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
})