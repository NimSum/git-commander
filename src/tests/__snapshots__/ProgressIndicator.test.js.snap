// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Indicator Should render card based on current round 1`] = `
<div
  className="progress-indicator"
>
  <span>
    Remote
  </span>
  <span />
  <span />
  <span
    className="current-round"
  />
  <span
    className="current-round"
  />
  <span
    className="current-round"
  >
    Local
  </span>
</div>
`;
